src/main/java/nl/tudelft/contextproject/Main.java

```diff
@@ -91,10 +121,13 @@ public static void setInstance(Main main) {
 	public boolean setController(Controller c) {
 		if (c != controller) {
 			if (controller != null) {
-				stateManager.detach(controller);
+				getStateManager().detach(controller);
 			}
 			controller = c;
-			stateManager.attach(controller);
+			
+			if (controller != null) {

```

Comment:
on line 122 c is checked to not be null, on line 126 the controller is set to c, on line 128 the controller is checked (which has just been set to c, which has already been checked)